shared/src/commonTest/kotlin/co/nimblehq/avishek/phong/kmmic/data.repository/SurveyRepositoryTest.kt

```diff
@@ -0,0 +1,166 @@
+package co.nimblehq.avishek.phong.kmmic.data.repository
+
+import app.cash.turbine.test
+import co.nimblehq.avishek.phong.kmmic.data.local.datasource.SurveyLocalDataSource
+import co.nimblehq.avishek.phong.kmmic.data.local.model.SurveyRealmObject
+import co.nimblehq.avishek.phong.kmmic.data.local.model.toSurvey
+import co.nimblehq.avishek.phong.kmmic.data.remote.datasource.SurveyRemoteDataSource
+import co.nimblehq.avishek.phong.kmmic.data.remote.model.SurveyApiModel
+import co.nimblehq.avishek.phong.kmmic.data.remote.model.toSurvey
+import co.nimblehq.avishek.phong.kmmic.domain.repository.SurveyRepository
+import io.kotest.matchers.shouldBe
+import io.mockative.Mock
+import io.mockative.any
+import io.mockative.classOf
+import io.mockative.given
+import io.mockative.mock
+import io.mockative.times
+import io.mockative.verify
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.test.runTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+
+@ExperimentalCoroutinesApi
+class SurveyRepositoryTest {
+
+    @Mock
+    private val mockSurveyRemoteDataSource = mock(classOf<SurveyRemoteDataSource>())
+    @Mock
+    private val mockSurveyLocalDataSource = mock(classOf<SurveyLocalDataSource>())
+
+    private lateinit var repository: SurveyRepository
+
+    private val mockThrowable = Throwable("mock")
+    private val mockSurveyRealmObject = SurveyRealmObject()
+        .apply { id = "id" }
+    private val mockSurveyApiModel = SurveyApiModel(
+        "id",
+        "type",
+        "title",
+        "description",
+        true,
+        "coverImageUrl",
+    )
+
+    @BeforeTest
+    fun setUp() {
+        repository = SurveyRepositoryImpl(mockSurveyRemoteDataSource, mockSurveyLocalDataSource)
+    }
+
+    @Test
+    fun `when there is no cached data and get surveys is succeeded, it returns surveys one time`() =
+        runTest {
+            given(mockSurveyLocalDataSource)
+                .function(mockSurveyLocalDataSource::getSurveys)
+                .whenInvoked()
+                .thenReturn(listOf())
```

Comment:
`emptyLIst()` would be more expressive IMO.
```suggestion
                .thenReturn(emptyList())
```